src/js/media-chrome-listbox.js

```diff
@@ -0,0 +1,416 @@
+  }
+
+  get listboxElement() {
+    return this.shadowRoot?.getElementById('listbox');
+  }
+
+  get listItemElements() {
+    return (
+      Array.from(this.listboxElement.querySelectorAll('[role="option"]')) ?? []
+    );
+  }
+
+  get activeDescendant() {
+    return (
+      this.listboxElement.getAttribute('aria-activedescendant') ?? undefined
+    );
+  }
+
+  set activeDescendant(value) {
+    if (value === this.activeDescendant) return;
+
+    if (this.activeElement) {
+      this.activeElement.classList.remove('focused');
+    }
+
+    if (isNil(value)) {
+      this.listboxElement.removeAttribute('aria-activedescendant');
+    } else {
+      this.listboxElement.setAttribute('aria-activedescendant', value);
+    }
+
+    if (this.activeElement) {
+      this.activeElement.classList.add('focused');
+      this.updateScroll();
+    }
+  }
+
+  get activeElement() {
+    return this.shadowRoot.getElementById(this.activeDescendant);
+  }
+
+  set activeElement(element) {
+    if (element === this.activeElement) return;
+    this.activeDescendant = element?.id;
+  }
+
+  get activeIndex() {
+    const idx = this.listItemElements.indexOf(this.activeElement);
+    return idx <= 0 ? idx : undefined;
+  }
+
+  set activeIndex(value) {
+    if (value === this.activeIndex) return;
+    if (!(isNil(value) || typeof value == 'number')) {
+      console.error(
+        `invalid value ${value} for ${getClassName(
+          this
+        )} activeIndex property. Must be a number, null, or undefined`
+      );
+      return;
+    }
+    const { listItemElements } = this;
+    if (
+      typeof value === 'number' &&
+      (value < 0 || value > listItemElements.length - 1)
+    ) {
+      console.error(
+        `invalid value ${value} for ${getClassName(
+          this
+        )} activeIndex property. Must be within the index range of data (0-${
+          listItemElements.length - 1
+        })`
+      );
+      return;
+    }
+    this.activeElement = listItemElements[value];
+  }
+
+  get selectedElement() {
+    return this.listboxElement.querySelector('[aria-selected]') ?? undefined;
+  }
+
+  set selectedElement(element) {
+    if (element === this.selectedElement) return;
+    this.selectedElement?.removeAttribute('aria-selected');
+    element?.setAttribute('aria-selected', 'true');
+  }
+
+  get selectedValue() {
+    return this.selectedElement?.id ?? undefined;
+  }
+
+  set selectedValue(value) {
+    if (value === this.selectedValue) return;
+    this.selectedElement = this.shadowRoot.getElementById(value);
+  }
+
+  render() {
+    const { selectedValue } = this;
+    const listItemEls = Array.from(this.data).map((value) => {
+      const listItemEl = document.createElement(this.itemRenderer);
+      listItemEl.setAttribute('role', 'option');
+      listItemEl.id = value;
+      listItemEl.innerText = this.format(value);
+      if (value === selectedValue) {
+        this.selectedElement = listItemEl;
```

Comment:
this is currently here to handle cases where data is updated/items are re-rendered after an item has already been selected.